core/trashcan.js

```diff
@@ -257,6 +258,13 @@ Blockly.Trashcan.prototype.left_ = 0;
  */
 Blockly.Trashcan.prototype.top_ = 0;
 
+/**
+ * Whether this has been initialized.
+ * @type {boolean}
+ * @private
+ */
+Blockly.Trashcan.prototype.initialized_ = false;
+
```

Comment:
I don't think this should be added to the interface (because it's private), but if we had an abstract class then I think it would be a reasonable add.
I've added this field so that there's no change in behavior after removing `this.vericalSpacing_`

Reply:
Ok, I was thinking we could make the `initialized_` property public and then check whether a plugin is initialized in workspace_svg before calling plugin.position(). However, looking at this a bit more I don't see many plugins position method being called before they are initialized. The reason we need this for trashcan and zoom controls is because we do the following: create the object --> call `svgResize` (nothing happens because it is not initialized)--> initialize the object --> call `svgResize` (object is actually positioned). This entire flow seems a bit suspicious and we should look into this more.